Neos.Media/Classes/Domain/Model/Dto/AssetResourceReplaced.php

```diff
@@ -0,0 +1,18 @@
+<?php
+namespace Neos\Media\Domain\Model\Dto;
+
+use Neos\Flow\ResourceManagement\PersistentResource;
+use Neos\Media\Domain\Model\AssetInterface;
+
+/**
+ *
+ */
+final readonly class AssetResourceReplaced
+{
+    public function __construct(
+        public AssetInterface $asset,
+        public PersistentResource $previousResource,
+        public PersistentResource $newResource
+    )
+    {}
```

Comment:
For now we agreed to use this pattern FYI ^^ https://github.com/neos/neos-development-collection/pull/4605#discussion_r1358113177

```suggestion
    ) {
    }
```